jdaviz/configs/specviz/plugins/viewers.py

```diff
@@ -365,118 +355,105 @@ def add_data(self, data, color=None, alpha=None, **layer_state):
         # trace representing the spectrum itself.
         result = super().add_data(data, color, alpha, **layer_state)
 
-        # Index of spectrum trace plotted by the super class. It is the
-        # **latest** item in figure.marks that is a Line instance
-        for ind, mark in reversed(list(enumerate(self.figure.marks))):
-            # we'll use __class__.__name__ since other entries (spectral lines,
-            # etc) defined in jdaviz.core.marks inherit from Lines
-            if mark.__class__.__name__ == 'Lines':
-                self.data_trace_pointer = ind
-                break
-        else:  # pragma: no cover
-            raise ValueError("could not find mark for added data")
-
-        self.error_trace_pointer = None
-        self.mask_trace_pointer = None
-
-        # Color and opacity are taken from the already plotted trace,
-        # in case they are not set explicitly by the caller.
-        self._color = self.figure.marks[self.data_trace_pointer].colors[0]
-        if color:
-            self._color = color
-
-        self._alpha = self.figure.marks[self.data_trace_pointer].opacities[0]
-        if alpha:
-            self._alpha = alpha
-
-        # An opacity defined specifically for the shaded areas.
-        self._alpha_shade = self._alpha / 3
-
```

Comment:
does removing this have consequences in any other situation that might not be covered by CI?

Reply:
Searched the whole code base and these attributes don't seem to appear elsewhere, so hopefully its safe 🤞